src/adapters/middleware-adapter.ts

```diff
@@ -40,31 +42,30 @@ export async function handler(event: CloudFrontRequestEvent): Promise<CloudFront
 
   // Process request
   const response = await index.default(nodeRequest, {
-    waitUntil: () => { },
+    waitUntil: () => {},
   });
   console.log("middleware response header", response.headers);
 
   // WORKAROUND: Pass headers from middleware function to server function (AWS specific) — https://github.com/serverless-stack/open-next#workaround-pass-headers-from-middleware-function-to-server-function-aws-specific
   if (response.headers.get("x-middleware-next") === "1") {
-    headers["x-op-middleware-request-headers"] = [{
-      key: "x-op-middleware-request-headers",
-      value: getMiddlewareRequestHeaders(response),
-    }];
-    headers["x-op-middleware-response-headers"] = [{
-      key: "x-op-middleware-response-headers",
-      value: getMiddlewareResponseHeaders(response),
-    }];
+    headers["x-op-middleware-request-headers"] = [
+      {
+        key: "x-op-middleware-request-headers",
+        value: getMiddlewareRequestHeaders(response),
+      },
+    ];
+    headers["x-op-middleware-response-headers"] = [
+      {
+        key: "x-op-middleware-response-headers",
+        value: getMiddlewareResponseHeaders(response),
+      },
+    ];
     return request;
   }
 
-  // In the case where `middleware.ts` response w/ a body, send that back to the client
-  let responseBody
-  const state = Object.getOwnPropertySymbols(response).find(s => s.toString() === 'Symbol(state)')
-  if (state) responseBody = response[state]?.body?.source
-
   return {
     status: response.status,
-    body: responseBody,
+    body: response.body ? await response.text() : undefined,
```

Comment:
👍

Reply:
yeah.. I will make a separate PR for that today. I want to add a DEBUG flag, and when u do `DEBUG=true open-next build`, it will `console.log`.